lms/static/scripts/frontend_apps/components/LaunchErrorDialog.js

```diff
@@ -0,0 +1,176 @@
+import { createElement } from 'preact';
+import { useRef } from 'preact/hooks';
+
+import Button from './Button';
+import Dialog from './Dialog';
+import ErrorDisplay from './ErrorDisplay';
+
+/**
+ * @typedef {import("preact").ComponentChildren} Children
+ * @typedef {import('./BasicLtiLaunchApp').ErrorState} ErrorState
+ */
+
+/**
+ * Common structure for LTI launch error dialogs.
+ *
+ * This is a "full-screen", non-cancelable dialog consisting of:
+ *
+ *  - An explanation of what went wrong and how to resolve it
+ *  - An optional "Try again" button
+ *  - An optional detailed error message for use when contacting support
+ *
+ * @param {Object} props
+ * @param {boolean} props.busy
+ * @param {Children} props.children
+ * @param {Error|null} [props.error]
+ * @param {() => void} [props.onRetry]
```

Comment:
You can use `VoidFunction` if you wish.

Reply:
The `VoidFunction` interface appears to be an interface that is defined in web API specs and is used by things like `queueMicrotask`. I don't think it is intended for general use outside of that, but please let me know if you've  seen/heard otherwise.